backend/review/views.py

```diff
@@ -129,30 +134,59 @@ def extra_metrics_section_filters_pcr(current_semester=None):
 )
 @permission_classes([IsAuthenticated])
 def course_reviews(request, course_code, semester=None):
+    request_semester = request.GET.get("semester")
+    topic_id = cache.get(course_code)
+    if topic_id is None:
+        try:
+            recent_course = most_recent_course_from_code(course_code, request_semester)
+        except Course.DoesNotExist:
+            raise Http404()
+        topic = recent_course.topic
+        course_id_list = list(topic.courses.values_list("id"))
+        topic_id = ".".join([str(id[0]) for id in sorted(course_id_list)])
+        cache.set(course_code, topic_id, MONTH_IN_SECONDS)
```

Comment:
why do we set the cache manually?

Reply:
instead of using the decorator

Reply:
also we shpuld make sure this cache is cleared every time we reimport and/or that the TTL is short enough.

Reply:
Cache is set manually because we want some control over setting the course to topic mappings and the topic to response mappings. Using the decorator, since the routes are based on course, would only allow for course to response mapping, which can be a bit space inefficient.

Reply:
Yup the relevant cache entries are cleared in the precompute_pcr_reviews script.